src/PackageParser.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
+ *
+ * This source code is licensed under the Apache 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { readJsonSync } from 'fs-extra';
+
+/**
+ * The PackageParser class.
+ * It provides methods to extract information form a package.json file.
+ */
+class PackageParser {
+  /**
+   * Constructor of the PackageParser
+   */
+  constructor() {
+    this.content = null;
+    this.linkableDependencies = [];
+  }
+
+  /**
+   * Initializes the package parser by parsing the package.json at the specified path.
+   * @param {string} path The path of the package.
+   * @return {PackageParser}
+   */
+  parse(path) {
+    try {
+      this.content = readJsonSync(`${path}/package.json`);
+    } catch (e) {
+      this.content = null;
+      throw new Error(`Invalid package at ${path}`);
+    }
+
+    return this;
+  }
+
+  /**
+   * Gets the name of the package.
+   * @return {string}
+   */
+  getName() {
+    if (this.content === null) {
+      throw new Error('PackageParser not initialized');
+    }
+
+    return this.content.name;
+  }
+
+  /**
+   * Sets a list of packages for dependency linking.
+   * @param {Array} [packages] The linkable packages.
```

Comment:
I don't think that the `packages` parameter really is optional, is it?

Reply:
I removed brackets, since i didn't apply a default value to the `packages` parameter